### PR TITLE
chore: Add Migration scripts along with extensions test

### PR DESCRIPTION
**Pull request by Lakshmipathi**
After building AMI, Perform extension install test. If tests passed, push migrations files to s3. Also create migration_output.sql file.
Signed-off-by: Lakshmipathi <lakshmi@supabase.io>

### CODE REVIEW

**Comment by Lakshmipathi** [comment]
Extension test passed. https://github.com/supabase/postgres/actions/runs/3419028280/jobs/5692036378  (ignore the s3 failure, its due to staging key rotation) 

pljava test passed after switching to `supabase_admin` instead of `postgres` though.

File: .github/workflows/ami-release.yml [hunk 1]
```diff
@@ -17,3 +17,17 @@ jobs:
       - name: Build AMI
         run: |
           packer build -var-file="development-arm.vars.pkr.hcl" -var-file="common.vars.pkr.hcl" amazon-arm64.pkr.hcl
+
+      - name: Merging migration files
+        run: cat $(ls -1) > ../migration-output.sql
+        working-directory: ${{ github.workspace }}/migrations/db/migrations
+
+      - name: Push migration files to S3
+        uses: jakejarvis/s3-sync-action@master
```
Comment (sweatybridge) [inline .github/workflows/ami-release.yml:26]:
```suggestion
        uses: jakejarvis/s3-sync-action@master
        with:
          args: --delete
```

Might want to replace existing files

File: .github/workflows/ami-release.yml [hunk 2]
```diff
@@ -17,3 +17,17 @@ jobs:
       - name: Build AMI
         run: |
           packer build -var-file="development-arm.vars.pkr.hcl" -var-file="common.vars.pkr.hcl" amazon-arm64.pkr.hcl
+
+      - name: Merging migration files
+        run: cat $(ls -1) > ../migration-output.sql
+        working-directory: ${{ github.workspace }}/migrations/db/migrations
```
Comment (sweatybridge) [inline .github/workflows/ami-release.yml:23]: Probably not needed anymore if we use migrate.sh

**Review by sweatybridge — APPROVED** [review]

**Review by darora — APPROVED** [review]

File: .github/workflows/ami-release.yml (hunk 1 above)
Comment (Lakshmipathi) [reply .github/workflows/ami-release.yml:26]: ok, added.

File: .github/workflows/ami-release.yml (hunk 2 above)
Comment (sweatybridge) [reply .github/workflows/ami-release.yml:23]: discussed offline: concatenating migrations into a single file could introduce unexpected behaviour as session configs, ie. `set [session | local] ...`, are persistent for the whole file when executed by psql. Will need to update `restore.ts` in a separate PR to run each migration individually.

**Pull request merged**